### ISSUE TITLE
Palmetto failed to power-on host

### ISSUE THREAD

**Issue by adamliyi** [openbmc/openbmc#1262]
I tested openbmc using latest auto-build image:
https://openpower.xyz/job/openbmc-build/distro=ubuntu,target=palmetto/lastBuild/artifact/images/palmetto/

The openbmc version is:

```
root@palmetto:~# cat /etc/os-release 
ID="openbmc-phosphor"
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="v1.99.3-42"
VERSION_ID="v1.99.3-42-ge8c604b"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) v1.99.3-42"
BUILD_ID="v1.99.3"
```

Run `obmcutil poweron` fails to power-on the host - watchdog will reset the booting.

```
Mar 07 06:47:35 palmetto systemd[1]: Started Start Power0.
Mar 07 06:47:35 palmetto system_manager.py[769]: Running System State: HOST_POWERING_ON
Mar 07 06:47:35 palmetto timemanager[854]:  PGOOD has changed..
Mar 07 06:47:37 palmetto kernel: ftgmac100 1e660000.ethernet eth0: NCSI interface up
Mar 07 06:47:37 palmetto kernel[705]: ftgmac100 1e660000.ethernet eth0: NCSI interface up
Mar 07 06:47:37 palmetto system_manager.py[769]: Running System State: HOST_POWERED_ON
Mar 07 06:47:37 palmetto timemanager[854]:  User Settings have changed..
Mar 07 06:47:37 palmetto power_control.exe[819]: PowerControl: pgood: 1, setting reset USB_RESET to 1
Mar 07 06:47:37 palmetto systemd[1]: Started Wait for Power0 to turn on.
Mar 07 06:47:37 palmetto timemanager[854]:  PGOOD has changed..
Mar 07 06:47:37 palmetto timemanager[854]: Changing time settings is *deferred* now
Mar 07 06:47:37 palmetto systemd[1]: Reached target Power0 On.
Mar 07 06:47:37 palmetto systemd[1]: Reached target Power0 (On).
Mar 07 06:47:37 palmetto systemd[1]: Starting Start Host0...
Mar 07 06:47:37 palmetto systemd[1]: Starting PCIE Slot Presence Detection For Chassis0...
Mar 07 06:47:37 palmetto systemd[1]: Reached target Chassis0 (Power On).
Mar 07 06:47:37 palmetto phosphor-chassis-state-manager[823]: Recieved signal that power ON is complete
Mar 07 06:47:37 palmetto phosphor-chassis-state-manager[823]: Change to Chassis Power State
Mar 07 06:47:38 palmetto systemd[1]: Started PCIE Slot Presence Detection For Chassis0.
Mar 07 06:47:38 palmetto control_host.exe[808]: Booting host
Mar 07 06:47:38 palmetto systemd[1]: Started Start Host0.
Mar 07 06:47:39 palmetto systemd[1]: Reached target Start Host0.
Mar 07 06:47:39 palmetto systemd[1]: Reached target Host0 (Started).
Mar 07 06:47:39 palmetto systemd[1]: Starting Start Watchdog0...
Mar 07 06:47:39 palmetto host_watchdog.exe[706]: Setting watchdog poll interval to: 30000
Mar 07 06:47:39 palmetto control_host.exe[808]: Using primary side of the bios flash
Mar 07 06:47:39 palmetto host_watchdog.exe[706]: Starting watchdog with poll interval: 30000
Mar 07 06:47:39 palmetto systemd[1]: Started Start Watchdog0.
Mar 07 06:47:39 palmetto systemd[1]: Reached target Chassis0 (Start).
Mar 07 06:47:39 palmetto phosphor-host-state-manager[807]: Recieved signal that host is running
Mar 07 06:47:39 palmetto phosphor-host-state-manager[807]: Change to Host State
Mar 07 06:47:40 palmetto system_manager.py[769]: Running System State: HOST_BOOTING
Mar 07 06:47:40 palmetto host_watchdog.exe[706]: Setting watchdog poll interval to: 30000
Mar 07 06:47:40 palmetto host_watchdog.exe[706]: Starting watchdog with poll interval: 30000
Mar 07 06:48:10 palmetto host_watchdog.exe[706]: Polling watchdog: 0
Mar 07 06:48:10 palmetto chassis_control.py[681]: Watchdog Error, Going to quiesce
Mar 07 06:48:10 palmetto systemd[1]: Reached target Quiesce Target.
Mar 07 06:48:11 palmetto phosphor-host-state-manager[807]: Auto reboot enabled. Beginning reboot...
Mar 07 06:48:11 palmetto phosphor-host-state-manager[807]: Host State transaction request
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Quiesce Target.
Mar 07 06:48:11 palmetto phosphor-host-state-manager[807]: Auto reboot enabled. Beginning reboot...
Mar 07 06:48:11 palmetto phosphor-host-state-manager[807]: Host State transaction request
Mar 07 06:48:11 palmetto systemd[1]: Starting Stop Watchdog0...
Mar 07 06:48:11 palmetto systemd[1]: Started Stop Power0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Chassis0 (Start).
Mar 07 06:48:11 palmetto systemd[1]: Stopped Start Watchdog0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped PCIE Slot Presence Detection For Chassis0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Chassis0 (Power On).
Mar 07 06:48:11 palmetto host_watchdog.exe[706]: Stopping watchdog
Mar 07 06:48:11 palmetto phosphor-chassis-state-manager[823]: Recieved signal that power ON is complete
Mar 07 06:48:11 palmetto power_control.exe[819]: PowerControl: setting power up POWER_PIN to 1
Mar 07 06:48:11 palmetto phosphor-chassis-state-manager[823]: Change to Chassis Power State
Mar 07 06:48:12 palmetto system_manager.py[769]: Running System State: HOST_POWERING_OFF
Mar 07 06:48:12 palmetto system_manager.py[769]: Running System State: HOST_POWERED_OFF
Mar 07 06:48:12 palmetto sensor_manager2.py[703]: Deleting OCC device
Mar 07 06:48:12 palmetto power_control.exe[819]: PowerControl: pgood: 0, setting reset USB_RESET to 0
Mar 07 06:48:12 palmetto timemanager[854]:  PGOOD has changed..
Mar 07 06:48:12 palmetto timemanager[854]:  User Settings have changed..
Mar 07 06:48:12 palmetto timemanager[854]:  PGOOD has changed..
Mar 07 06:48:12 palmetto timemanager[854]: Changing time settings allowed now
Mar 07 06:48:11 palmetto systemd[1]: Starting Wait for Power0 to turn off...
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Host0 (Started).
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Start Host0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped Start Host0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Power0 (On).
Mar 07 06:48:11 palmetto systemd[1]: Stopped target Power0 On.
Mar 07 06:48:11 palmetto systemd[1]: Stopped Start Power0.
Mar 07 06:48:11 palmetto systemd[1]: Stopped Wait for Power0 to turn on.
Mar 07 06:48:11 palmetto systemd[1]: Started Stop Watchdog0.
Mar 07 06:48:12 palmetto phosphor-host-state-manager[807]: Recieved signal that host is running
Mar 07 06:48:12 palmetto phosphor-host-state-manager[807]: Change to Host State
Mar 07 06:48:12 palmetto systemd[1]: Started Wait for Power0 to turn off.
Mar 07 06:48:13 palmetto systemd[1]: Reached target Chassis0 (Power Off).
Mar 07 06:48:13 palmetto phosphor-chassis-state-manager[823]: Recieved signal that power OFF is complete
Mar 07 06:48:13 palmetto phosphor-chassis-state-manager[823]: Change to Chassis Power State
Mar 07 06:48:13 palmetto systemd[1]: Reached target Power0 Off.
Mar 07 06:48:13 palmetto systemd[1]: Reached target Chassis0 (Stop).
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Recieved signal that host is off
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Change to Host State
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Reached intermediate state, going to next
Mar 07 06:48:13 palmetto systemd[1]: Starting Wait for Power0 to turn on...
Mar 07 06:48:13 palmetto systemd[1]: Stopped target Chassis0 (Stop).
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Recieved signal that host is off
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Change to Host State
Mar 07 06:48:13 palmetto phosphor-host-state-manager[807]: Reached intermediate state, going to next
Mar 07 06:48:13 palmetto systemd[1]: Stopped Stop Watchdog0.
Mar 07 06:48:13 palmetto systemd[1]: Stopped target Chassis0 (Power Off).
Mar 07 06:48:13 palmetto phosphor-chassis-state-manager[823]: Recieved signal that power OFF is complete
Mar 07 06:48:13 palmetto phosphor-chassis-state-manager[823]: Change to Chassis Power State
Mar 07 06:48:13 palmetto systemd[1]: Stopped target Power0 Off.
Mar 07 06:48:13 palmetto systemd[1]: Stopped Wait for Power0 to turn off.
Mar 07 06:48:13 palmetto systemd[1]: Stopped Stop Power0.
Mar 07 06:48:13 palmetto systemd[1]: Starting Start Power0...
Mar 07 06:48:14 palmetto power_control.exe[819]: PowerControl: setting power up POWER_PIN to 0
Mar 07 06:48:14 palmetto systemd[1]: Started Start Power0.
Mar 07 06:48:14 palmetto kernel: ftgmac100 1e660000.ethernet eth0: NCSI interface down
Mar 07 06:48:14 palmetto kernel[705]: ftgmac100 1e660000.ethernet eth0: NCSI interface down
```

@gwshan is seeing the same issue.

**Comment by gkeishin** [comment]
Apart from ftgmac100  issue seen https://github.com/openbmc/openbmc/issues/1064
There is this Host going for auto reboot reaching `Quiesce Mode` 
```
Mar 07 06:48:10 palmetto chassis_control.py[681]: Watchdog Error, Going to quiesce
Mar 07 06:48:10 palmetto systemd[1]: Reached target Quiesce Target.
```

**Comment by shenki** [comment]
fwiw, I boot the host with the latest master on palmetto daily

**Comment by adamliyi** [comment]
Just check the Palmetto (palm309) in the reservation system, which have the same issue - so I think this bug is valid.

```
root@palmetto:~# cat /etc/os-release 
ID="openbmc-phosphor"
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="v1.99.3-45"
VERSION_ID="v1.99.3-45-g8e6b232"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) v1.99.3-45"
BUILD_ID="v1.99.3"
```

**Comment by geissonator** [comment]
I've seen this a bit lately on our other systems.  Looks to me like PNOR is getting corrupted or something to the point where hostboot can't start.  Try re-flashing PNOR?  That fixes it for me.

**Comment by gwshan** [comment]
@adamliyi helped build BMC image from "master" branch. The host doesn't boot after "obmcutil poweron". The host boots well after I reflashed the PNOR image.

**Comment by gwshan** [comment]
@adamliyi I think this bug can be closed if we doesn't plan to track the corrupted PNOR issue by this.

**Comment by yadron** [comment]
The same situation with Barreleye (cannot boot host), and I noted an error during update (when tried updating second time for the same latest master image):

```
root@barreleye:/run/initramfs# ./update 
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32336/32768 (98%) flashcp: verification mismatch at 0x1f94000
```

**Comment by yadron** [comment]
For me it looks like some processes uses incorrectly flash memory and/or CPU (too often read? too often write? Could be some broken processes that being restarted in cycle).

After prev. msg, I've recompiled image and tried updating again (every time different errors during verification - see below-, so for me it looks like non-flash hardware problem, but software problem, and at last I think that some new python or dbus services or sensors readers makes CPU 100% busy from time to time for the latest master bundle and as result flashcp works badly and host cannot boot maybe because of this too or because of ipmid broken):

```
root@barreleye:/run/initramfs# ./update 
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32392/32768 (98%) flashcp: verification mismatch at 0x1fa2000

```
And one more attempt:

```
root@barreleye:/run/initramfs# ./update 
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32404/32768 (98%) flashcp: verification mismatch at 0x1fa5000
root@barreleye:/run/initramfs#
```

More attempts.. Every time verify does more than prev attampt, but fails on ~ +20 kb blocks..

I've tried killing "/usr/sbin/flash_bios.exe" process (just in case it does something with flash).

Updating again, the same error.

I've tried killing bmc_update.py then:

```
root@barreleye:/run/initramfs# cp /usr/sbin/bmc_update.py /usr/sbin/bmc_update_.py
root@barreleye:/run/initramfs# rm /usr/sbin/bmc_update.py                         
root@barreleye:/run/initramfs# ps|grep bmc_up
 5523 root       0:03 python /usr/sbin/bmc_update.py
 5564 root       0:00 grep bmc_up
root@barreleye:/run/initramfs# kill 5523
root@barreleye:/run/initramfs# ps|grep bmc_up
 5569 root       0:00 grep bmc_up
root@barreleye:/run/initramfs# ./update


```

and again.. Still verify error..

I've noted that for some reason during flash update or any other activity the system stalls sometimes (frozen for say 20 seconds without reason).. So I thought that this is not flash error, but some services makes system busy and as result flashcp works incorrectly because of this..

So tried "top" few times and still removing most active services:

```
root@barreleye:/run/initramfs# cp /usr/sbin/hwmon.py /usr/sbin/hwmon_.py  
root@barreleye:/run/initramfs# rm /usr/sbin/hwmon.py                      
root@barreleye:/run/initramfs# ps|grep hwmo
  715 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/rtc@68
  718 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/lm75@4a
  778 root       5:27 python /usr/sbin/hwmon.py
 5800 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2e
 5808 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2d
 5823 root       0:00 grep hwmo
root@barreleye:/run/initramfs# kill 778
root@barreleye:/run/initramfs# ps|grep hwmo
  715 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/rtc@68
  718 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/lm75@4a
 5800 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2e
 5808 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2d
 5836 root       0:00 grep hwmo
root@barreleye:/run/initramfs# ps|grep hwmo
  715 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/rtc@68
  718 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/lm75@4a
 5800 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2e
 5808 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2d
root@barreleye:/run/initramfs# ps|grep hwmo
  715 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/rtc@68
  718 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/lm75@4a
 5800 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2e
 5808 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2d
 5840 root       0:00 grep hwmo
root@barreleye:/run/initramfs# ps|grep hwmo
  715 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/rtc@68
  718 root       0:01 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@40/lm75@4a
 5800 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2e
 5808 root       0:02 /usr/sbin/phosphor-hwmon-readd -o /ahb/apb/i2c@1e78a000/i2c-bus@1c0/nct7904@2d
 5842 root       0:00 grep hwmo
root@barreleye:/run/initramfs# cp /usr/sbin/phosphor-hwmon-readd /usr/sbin/phosphor-hwmon-readd_
root@barreleye:/run/initramfs# rm /usr/sbin/phosphor-hwmon-readd                                
root@barreleye:/run/initramfs# kill 715
root@barreleye:/run/initramfs# kill 718
root@barreleye:/run/initramfs# kill 5800
root@barreleye:/run/initramfs# kill 5808
root@barreleye:/run/initramfs# ps|grep hwm
 5890 root       0:00 grep hwm
root@barreleye:/run/initramfs#

```
And, yes! It updated successfully! Look:

```
root@barreleye:/run/initramfs# ./update 
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32768/32768 (100%)

```

After all this, trying again:

```
root@barreleye:/run/initramfs# reboot
-sh: /sbin/reboot: Input/output error
```

Ok, let's do hardware reboot.

After hardware reboot:

root@barreleye:~# obmcutil state
 = BMC_READY

root@barreleye:~# obmcutil poweron

root@barreleye:~# obmcutil state
 = HOST_BOOTING

Actually, host booting fails.

As I already know, some processes freezes BMC (CPU 100% and stalls for up to 30secs), and both I know that during boot some IPMI commands execution required. Looks like ipmid fails during boot or some-kind of other fail that makes overall BMC unstable.

Ok, let's kill top services one after other again (as long as system still stalls)...

Killing top services didn't help, but I noted that ipmid fails (and I know that if it does, then the host cannot boot):

_root@barreleye:~# ps|grep ipmid
  695 root       0:00 /usr/sbin/netipmid
 2024 root       0:00 /usr/sbin/ipmid

root@barreleye:~# ipmid 
Registering handler:[/usr/lib/host-ipmid/libsysintfcmds.so.0]
Registering NetFn:[0x6], Cmd:[0x35]
...
Registering NetFn:[0x2C], Cmd:[0x0]_
**Failed to get /org/openbmc/settings/host0 busname: No route to host**

Looking to this error here
./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/git/*.cpp

I already know how to debug this, so found that this is the line 277 of ipmid.cpp:

phosphor-ipmi-host/1.0-r1/git/ipmid.cpp
```
267 void cache_restricted_mode()
268 {
269     sd_bus *bus = ipmid_get_sd_bus_connection();
270     sd_bus_message *reply = NULL;
271     sd_bus_error error = SD_BUS_ERROR_NULL;
272     int rc = 0;
273     char  *busname = NULL;
274 
275     rc = mapper_get_service(bus, settings_host_object, &busname);
276     if (rc < 0) {
277         fprintf(stderr, "Failed to get %s busname: %s\n",
```

and this one being called from main:

478 int main(int argc, char *argv[])
479 {
480     int r;
...
536     // Initialise restricted mode
537     cache_restricted_mode();

- so you see the reason of ipmid frozen (so it doesn't answer petitboot/etc during boot and host cannot boot).

As a workaround, let's comment line 537 (and both line 318) and then trying again:

```

[build@obmc build]$ rm ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/build/.libs/libapphandler.so.0
[build@obmc build]$ rm ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/build/ipmid
[build@obmc build]$ bitbake -f -c compile phosphor-ipmi-host
[build@obmc build]$ scp ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/build/ipmid root@172.17.27.82:/usr/sbin/ipmid_new

root@barreleye:~# rm /usr/sbin/ipmid

root@barreleye:~# ps|grep ipmid
  695 root       0:00 /usr/sbin/netipmid
 2024 root       0:00 /usr/sbin/ipmid

root@barreleye:~# kill 2024

root@barreleye:~# /usr/sbin/ipmid_new

```
- now ipmid works (but host not booted still).


Looks like I've broken something

(
```
root@barreleye:~# obmcutil state 
Traceback (most recent call last):
  File "/usr/sbin/obmcutil", line 196, in <module>
    obj = bus.get_object(bus_name, obj_path)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python2.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.openbmc.managers.System was not provided by any .service files

```
- is it a problem of "_/usr/sbin/system_manager.py at BMC that is "meta-phosphor/common/recipes-phosphor/system/obmc-mgr-system". This is DBUS component (that is "/etc/dbus-1/system.d/org.openbmc.managers.System.conf")._" or what?
), so let's reinstall image and try again then:

```

root@barreleye:/run/initramfs# ./update 
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32768/32768 (100%)
```

reboot
- cannot reboot (because system error reading FS:
```
"SQUASHFS error: Unable to read page, block c872c, size bcb4
SQUASHFS error: Unable to read data cache entry [c872c]
SQUASHFS error: Unable to read page, block c872c, size bcb4
SQUASHFS error: Unable to read data cache entry [c872c]
SQUASHFS error: Unable to read page, block c872c, size bcb4").
```

Both these files do this error:

```
root@barreleye:/run/initramfs# cat /bin/systemd-sysusers 
cat: read error: Input/output error

root@barreleye:/run/initramfs# cat /bin/systemctl 
cat: read error: Input/output error
```

Looking syslogs:

```
dmesg|tail

SQUASHFS error: Unable to read data cache entry [c872c]
SQUASHFS error: Unable to read page, block c872c, size bcb4
SQUASHFS error: Unable to read data cache entry [c872c]
SQUASHFS error: Unable to read page, block c872c, size bcb4
SQUASHFS error: Unable to read data cache entry [c872c]
SQUASHFS error: Unable to read page, block c872c, size bcb4
SQUASHFS error: Unable to read data cache entry [c872c]
```

Ok, I know where to get it:
```
[build@obmc build]$ scp ./tmp/work/armv5e-openbmc-linux-gnueabi/systemd/1_229+gitAUTOINC+714c62b463-r0/package/bin/systemctl root@172.17.27.83:/usr/sbin/
```

```
root@barreleye:/run/initramfs# which systemctl
/bin/systemctl
root@barreleye:/run/initramfs# 
root@barreleye:/run/initramfs# rm /bin/systemctl 
root@barreleye:/run/initramfs# ln /usr/sbin/systemctl /bin/
```

reboot
- this time Ok.

And still rebooting (for more than 15 mins).. Cannot connect to BMC anymore.. Let's check it tomorrow then, but not tomorrow, because it is alive again!!

Are you becoming interested of this long story?

Then read.

```
ssh root@172.17.27.84
The authenticity of host '172.17.27.84 (172.17.27.84)' can't be established.
RSA key fingerprint is 9d:39:08:9b:3a:b0:7c:6b:2d:a1:2f:83:d1:ef:77:f6.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '172.17.27.84' (RSA) to the list of known hosts.
root@172.17.27.84's password: 


root@barreleye:~# cat /etc/os-release 
ID="openbmc-phosphor"
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="v1.99.3-48"
VERSION_ID="v1.99.3-48-gf38bc1e"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) v1.99.3-48"
BUILD_ID="v1.99.3"
```

And no ipmid, so looks like it is broken:

```
root@barreleye:~# ps|grep ipmid
  716 root       0:00 /usr/sbin/netipmid
 1007 root       0:00 grep ipmid
```

and "obmcutil state" doesn't work still:

```
root@barreleye:~# obmcutil state
...
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.openbmc.managers.System was not provided by any .service files
```

and system_manager.py being restarted in cycle:

```
root@barreleye:~# ps|grep system_ma
 1172 root       0:03 python /usr/sbin/system_manager.py
 1225 root       0:00 grep system_ma
root@barreleye:~# 
root@barreleye:~# 
root@barreleye:~# ps|grep system_ma
 1234 root       0:00 grep system_ma
 1235 root       0:00 python /usr/sbin/system_manager.py
root@barreleye:~# 
root@barreleye:~# 
root@barreleye:~# ps|grep system_ma
 1235 root       0:00 python /usr/sbin/system_manager.py
 1243 root       0:00 grep system_ma
```

The reason is that ObjectMapper not avaliable:

```
root@barreleye:~# python /usr/sbin/system_manager.py
Running System State: BASE_APPS
Traceback (most recent call last):
  File "/usr/sbin/system_manager.py_", line 210, in <module>
    obj = SystemManager(bus, OBJ_NAME)
  File "/usr/sbin/system_manager.py_", line 55, in __init__
    self.SystemStateHandler(System.SYSTEM_STATES[0])
  File "/usr/sbin/system_manager.py_", line 74, in SystemStateHandler
    callback=self.gotoNextState)
  File "/usr/lib/python2.7/site-packages/obmc/mapper/utils.py", line 31, in __init__
    introspect=False)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python2.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name xyz.openbmc_project.ObjectMapper was not provided by any .service files
```

So looks like some problems with
/etc/dbus-1/system.d/xyz.openbmc_project.ObjectMapper.conf
that is phosphor-mapper.

And it works:

```
root@barreleye:~# which phosphor-mapper 
/usr/sbin/phosphor-mapper
root@barreleye:~# ps|grep phosphor-mapper
 1517 root       0:44 python /usr/sbin/phosphor-mapper

```

and all it does it runs

obmc.mapper.server.server_main()
(that is "import obmc.mapper.server").

I've just restarted phosphor-mapper and then /usr/sbin/system_manager.py started working:

```
root@barreleye:~# which phosphor-mapper 
/usr/sbin/phosphor-mapper
root@barreleye:~# ps|grep phosphor-mapper
 1517 root       0:44 python /usr/sbin/phosphor-mapper
 1687 root       0:00 grep phosphor-mapper
root@barreleye:~# kill 1517
root@barreleye:~# ps|grep phosphor-mapper
1689 root       1:14 python /usr/sbin/phosphor-mapper

```
...
and now system_manager.py working:

```
root@barreleye:~# python /usr/sbin/system_manager.py_ 
Running System State: BASE_APPS
SystemManager Init Done
Running SystemManager
SystemManager Goto System State: BMC_STARTING
Running System State: BMC_STARTING
SystemManager Goto System State: BMC_STARTING2
Running System State: BMC_STARTING2
SystemManager Goto System State: BMC_READY
Running System State: BMC_READY
```

and ipmid now works Ok:

```
root@barreleye:~# ps|grep ipmid
  716 root       0:00 /usr/sbin/netipmid
 1140 root       0:00 /usr/sbin/ipmid
```

```
root@barreleye:~# ipmid 
Registering handler:[/usr/lib/host-ipmid/libsysintfcmds.so.0]
Registering NetFn:[0x6], Cmd:[0x35]
Registering NetFn:[0x6], Cmd:[0x2E]
...
Registering NetFn:[0x2C], Cmd:[0x0]

Restricted mode = 0

```

and obmcutil works:

```
root@barreleye:~# obmcutil state
 = BMC_READY
```


but some problems with poweron:

```
root@barreleye:~# obmcutil poweron     
Traceback (most recent call last):
  File "/usr/sbin/obmcutil", line 196, in <module>
    obj = bus.get_object(bus_name, obj_path)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python2.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python2.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name xyz.openbmc_project.State.Host was not provided by any .service files
```

Need to fix
/etc/dbus-1/system.d/xyz.openbmc_project.State.Host.conf
this time... That is phosphor-state-manager
Let's check "phosphor-host-state-manager" and "phosphor-bmc-state-manager" and "phosphor-chassis-state-manager".

These two looks like Ok:
```
root@barreleye:~# ps|grep state-ma
  800 root       0:00 /usr/sbin/phosphor-chassis-state-manager
  802 root       0:00 /usr/sbin/phosphor-bmc-state-manager
```

Anyway, let's restart it:

```
root@barreleye:~# kill 800
root@barreleye:~# kill 802
root@barreleye:~# ps|grep state-ma
 1773 root       0:00 /usr/sbin/phosphor-chassis-state-manager
 1774 root       0:00 /usr/sbin/phosphor-bmc-state-manager
```

and this one started then:

```
root@barreleye:~# /usr/sbin/phosphor-host-state-manager &
[1] 1778
root@barreleye:~# 
root@barreleye:~# 
root@barreleye:~# ps|grep 1778    
 1778 root       0:00 /usr/sbin/phosphor-host-state-manager
```

So now

```
root@barreleye:~# ps|grep state-ma
 1773 root       0:00 /usr/sbin/phosphor-chassis-state-manager
 1774 root       0:00 /usr/sbin/phosphor-bmc-state-manager
 1778 root       0:00 /usr/sbin/phosphor-host-state-manager

```

and I can run poweron cmd:

`root@barreleye:~# obmcutil poweron`

```
root@barreleye:~# obmcutil bootprogress
units = 
value = Off
error = 0
root@barreleye:~# 
root@barreleye:~# 
root@barreleye:~# obmcutil poweroff
root@barreleye:~# obmcutil bootprogress
units = 
value = Off
error = 0
root@barreleye:~# obmcutil poweron 
root@barreleye:~# 
root@barreleye:~# 
root@barreleye:~# obmcutil bootprogress
units = 
value = Off
error = 0
root@barreleye:~# obmcutil state       
 = HOST_POWERING_ON

```
, but ipmid doesn't output anything:

-- is it because of "Failed to acquire service name: File exists" error? See:

```
root@barreleye:~# ipmid
...
Registering handler:[/usr/lib/host-ipmid/libhostservice.so.0]
Failed to acquire service name: File exists
```

so I uncommented cache_restricted_mode() here
vi ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/git/ipmid.cpp
and checked where "Failed to acquire service name: File exists"
./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-ipmi-host/1.0-r1/git/host-services.c
```
103 int start_host_service(sd_bus *bus, sd_bus_slot *slot)
104 {
...
120         /* Take one in OpenBmc */
121         rc = sd_bus_request_name(bus, "org.openbmc.HostServices", 0);
122         if (rc < 0)
123         {
124             fprintf(stderr, "Failed to acquire service name: %s\n", strerror(-rc));
125         }

```

Need to fix "org.openbmc.HostServices" this time...
No, don't need to fix it, but let's debug ipmid anyway.

Added some debug output to it, so it says:

```
root@barreleye:~# /usr/sbin/ipmid_new 
Registering handler:[/usr/lib/host-ipmid/libsysintfcmds.so.0]
Registering NetFn:[0x6], Cmd:[0x35]
Registering NetFn:[0x6], Cmd:[0x2E]
Registering NetFn:[0x6], Cmd:[0x31]
Registering handler:[/usr/lib/host-ipmid/libstrgfnhandler.so.1]
Registering NetFn:[0xA], Cmd:[0x12]
Registering handler:[/usr/lib/host-ipmid/liboemhandler.so.0]
Registering NetFn:[0x32], Cmd:[0xF0]
Registering NetFn:[0x32], Cmd:[0x10]
Registering handler:[/usr/lib/host-ipmid/libhostservice.so.0]
Registering handler:[/usr/lib/host-ipmid/libapphandler.so.0]
Registering NetFn:[0x6], Cmd:[0x36]
Registering NetFn:[0x6], Cmd:[0xFF]
Registering NetFn:[0x6], Cmd:[0x22]
Registering NetFn:[0x6], Cmd:[0x24]
Registering NetFn:[0x6], Cmd:[0x1]
Registering NetFn:[0x6], Cmd:[0x4]
Registering NetFn:[0x6], Cmd:[0x8]
Registering NetFn:[0x6], Cmd:[0x6]
Registering NetFn:[0x6], Cmd:[0x40]
Registering NetFn:[0x6], Cmd:[0x42]
Registering NetFn:[0x4], Cmd:[0xFF]
Registering NetFn:[0x4], Cmd:[0x2F]
Registering NetFn:[0x4], Cmd:[0x30]
Registering NetFn:[0x4], Cmd:[0x2D]
Registering NetFn:[0xA], Cmd:[0xFF]
Registering NetFn:[0xA], Cmd:[0x48]
Registering NetFn:[0xA], Cmd:[0x49]
Registering NetFn:[0xA], Cmd:[0x40]
Registering NetFn:[0xA], Cmd:[0x42]
Registering NetFn:[0xA], Cmd:[0x44]
Registering NetFn:[0x0], Cmd:[0xFF]
Registering NetFn:[0x0], Cmd:[0x0]
Registering NetFn:[0x0], Cmd:[0x9]
Registering NetFn:[0x0], Cmd:[0x1]
Registering NetFn:[0x0], Cmd:[0x2]
Registering NetFn:[0x0], Cmd:[0x8]
Registering NetFn:[0x2C], Cmd:[0x3]
Registering NetFn:[0xC], Cmd:[0xFF]
Registering NetFn:[0xC], Cmd:[0x1]
Registering NetFn:[0xC], Cmd:[0x2]
Registering NetFn:[0x6], Cmd:[0x2]
Registering NetFn:[0x6], Cmd:[0x3]
Registering NetFn:[0x2C], Cmd:[0x0]
Restricted mode = 0
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_wait
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_wait
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_wait
sd_bus_process
sd_bus_process returned > 0
sd_bus_process
sd_bus_wait

```

What is it stalled with "sd_bus_wait"?

Possibly (looking to
https://lists.freedesktop.org/archives/systemd-devel/2016-January/035485.html
) the main code of 

```
    for (;;) {
fprintf(stderr, "sd_bus_process\n");
        /* Process requests */
        r = sd_bus_process(bus, NULL);
        if (r < 0) {
            fprintf(stderr, "Failed to process bus: %s\n", strerror(-r));
            goto finish;
        }
        if (r > 0) {
fprintf(stderr, "sd_bus_process returned > 0\n");
            continue;
        }

fprintf(stderr, "sd_bus_wait\n");
        r = sd_bus_wait(bus, (uint64_t) - 1);
        if (r < 0) {
            fprintf(stderr, "Failed to wait on bus: %s\n", strerror(-r));
            goto finish;
        }
    }

```
 is incorrect.

I've changed timeout "-1" to 1 in sd_bus_wait, but nothing changed and looks like no changes in host-ipmid since prev build at all (and everything worked Ok in prev build).

The the 
poweron (xyz.openbmc_project.State.Host->RequestedHostTransition)
buggy itself maybe.

And there is a change from prev time:

```
diff ../../../openbmc-prev/openbmc/build/tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-dbus-interfaces/1.0-r1/build/xyz/openbmc_project/State/Host/server.cpp ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-dbus-interfaces/1.0-r1/build/xyz/openbmc_project/State/Host/server.cpp
293a294
>             std::make_tuple( "xyz.openbmc_project.State.Host.HostState.Quiesced",                 Host::HostState::Quiesced ),
diff ../../../openbmc-msi/openbmc/build/tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-dbus-interfaces/1.0-r1/build/xyz/openbmc_project/State/Host/server.hpp ./tmp/work/armv5e-openbmc-linux-gnueabi/phosphor-dbus-interfaces/1.0-r1/build/xyz/openbmc_project/State/Host/server.hpp
51a52
>             Quiesced,
```

What is "Quiesced"?

And
root@barreleye:~# obmc-console-client
not working too for this build (or just no output).

At last I thought that this build is very bad, so returning back to prev one (that was from 28feb2017):

`image-28feb2017$ scp ./image-bmc root@172.17.27.84:/run/initramfs/image-bmc`

Checking that it uploaded Ok:

```
root@barreleye:/run/initramfs# ls -l ./image-bmc 
-rw-r--r--    1 root     root      33554432 Mar  9 19:03 ./image-bmc
root@barreleye:/run/initramfs# 
root@barreleye:/run/initramfs# ./update 
update:
/dev/mtdblock4 /run/initramfs/ro squashfs ro,relatime 0 0
ERROR: Device mtd4 is mounted, image-bmc is busy.
root@barreleye:/run/initramfs# df
Filesystem           1K-blocks      Used Available Use% Mounted on
dev                     247708         0    247708   0% /dev
tmpfs                   248988     51300    197688  21% /run
/dev/mtdblock4           14976     14976         0 100% /run/initramfs/ro
/dev/mtdblock5            4096      1548      2548  38% /run/initramfs/rw
cow                       4096      1548      2548  38% /
tmpfs                   248988         0    248988   0% /dev/shm
tmpfs                   248988         0    248988   0% /sys/fs/cgroup
tmpfs                   248988         0    248988   0% /tmp
tmpfs                   248988         4    248984   0% /var/volatile
root@barreleye:/run/initramfs# umount /dev/mtdblock4 
root@barreleye:/run/initramfs# umount /dev/mtdblock5 

```

and running update:

```
root@barreleye:/run/initramfs# ./update
update:
Updating bmc...
Erasing block: 128/128 (100%) 
Writing kb: 32768/32768 (100%) 
Verifying kb: 32768/32768 (100%)
```
...
reboot
-sh: /sbin/reboot: Input/output error
, hmmmm,

again:

```
[build@obmc build]$ scp ./tmp/work/armv5e-openbmc-linux-gnueabi/systemd/1_229+gitAUTOINC+714c62b463-r0/package/bin/systemctl root@172.17.27.84:/usr/sbin/

root@barreleye:/run/initramfs# which systemctl
/bin/systemctl
root@barreleye:/run/initramfs# rm /bin/systemctl
root@barreleye:/run/initramfs# ln /usr/sbin/systemctl /bin/systemctl
```

Now Ok:

```
root@barreleye:/run/initramfs# reboot

Connection to 172.17.27.84 closed by remote host.
Connection to 172.17.27.84 closed.
```

Waited for about 20 mins for reboot (possibly because of read-from-FS problems, so it took much time for services termination).

Now got prev. master build (everything started and no read-flash errors):

```
root@barreleye:~# cat /etc/os-release 
ID="openbmc-phosphor"
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="v1.99.3-dirty"
VERSION_ID="v1.99.3-dirty"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) v1.99.3-dirty"
BUILD_ID="v1.99.3"

root@barreleye:~# ps|grep state-man
  832 root       0:00 /usr/sbin/phosphor-chassis-state-manager
  838 root       0:00 /usr/sbin/phosphor-bmc-state-manager
  840 root       0:00 /usr/sbin/phosphor-host-state-manager
  928 root       0:00 grep state-man

root@barreleye:~# vi /bin/systemctl 
:wq
root@barreleye:~#

root@barreleye:~# ps|grep system_mana
  757 root       0:13 python /usr/sbin/system_manager.py

```

Let's try starting host:

```
root@barreleye:~# obmcutil poweron

and ipmid output then:

Handling Netfn:[0x6], Cmd:[0x36]
IPMI SET ACPI STATE Ignoring for now
Failed to call the method: Connection timed out
Dest: :1.1, Path: /org/openbmc/HostIpmi/1
WATCHDOG SET Timer:[0x4B0] 100ms intervals
Failed to call the method: Input/output error
Dest: :1.1, Path: /org/openbmc/HostIpmi/1
WATCHDOG RESET
Failed to call the method: Input/output error
Dest: :1.1, Path: /org/openbmc/HostIpmi/1
...
No registered handlers for NetFn:[0x3A], Cmd:[0x1A] trying Wilcard implementation 
No Registered handlers for NetFn:[0x3A],Cmd:[0xFF]
ERROR:[0xC1] handling NetFn:[0x3A], Cmd:[0x1A]
IPMI SET_SYS_BOOT_OPTIONS reqptr->parameter =[5]
Failed to find Sensor Type 0x0
997: Default
Failed to issue method call: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/dbus/service.py", line 707, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/lib/python2.7/site-packages/obmc/dbuslib/bindings.py", line 82, in Set
    self.validator(interface_name, property_name, new_value)
  File "/usr/sbin/settings_manager.py", line 236, in input_validator
    self.validate_list(shk['allowed'], value)
  File "/usr/sbin/settings_manager.py", line 176, in validate_list
    raise ValueError("Invalid input. Data not in allowed values")
ValueError: Invalid input. Data not in allowed values

and console-client working again:

root@barreleye:~# obmc-console-client 
 38.70957|ISTEP 21. 1
....

```

and **host booted well** using "v1.99.3-dirty".



As result, the version "v1.99.3-48" looks very-very-very buggy (it looks like it even broken FS so it looks like flash-hardware problems instead, why? Is it because of CPU 100% busy like I thought above?).
If it will not be fixed soon, then I could suggest moving back to "v1.99.3-dirty" and trying aging...

**Comment by adamliyi** [comment]
@gwshan , @geissonator . Yes, I tried to re-flash PNOR, and Palmetto host can be booted correctly on my side too. I talked with FW people, it is said that the PNOR has the possibility to become corrupted - if the host is powered off during booting, since the PNOR will do Read and Write during booting.  Will close this issue.
